scripts/calculateRevenue/protocols/rhino/index.ts

```diff
@@ -0,0 +1,179 @@
+import { BridgeTransaction } from './types'
+import { fetchTokenPrices } from '../utils/tokenPrices'
+import { getTokenPrice } from '../beefy'
+import { paginateQuery } from '../../../utils/hypersyncPagination'
+import { Address, fromHex, isAddress, zeroAddress } from 'viem'
+import { getFirstBlockAtOrAfterTimestamp } from '../utils/events'
+
+export async function getUserBridges({
+  address,
+  contractAddress,
+  startTimestamp,
+  endTimestampExclusive,
+  client,
+  networkId,
+}: {
+  address: Address
+  contractAddress: Address
+  startTimestamp: Date
+  endTimestampExclusive: Date
+  client: HypersyncClient
+  networkId: NetworkId
+}): Promise<BridgeTransaction[]> {
+  const fromBlock = await getFirstBlockAtOrAfterTimestamp(
+    networkId,
+    startTimestamp,
+  )
+  const toBlock = await getFirstBlockAtOrAfterTimestamp(
+    networkId,
+    endTimestampExclusive,
+  )
+  const query = {
+    logs: [
+      { address: [contractAddress], topics: [[BRIDGED_WITHDRAWAL_TOPIC]] },
+    ],
+    fieldSelection: {
+      log: [LogField.BlockNumber, LogField.Data],
+    },
+    fromBlock,
+    toBlock,
+  }
+
+  const bridges: BridgeTransaction[] = []
+  await paginateQuery(client, query, async (response) => {
+    for (const bridge of response.data.logs) {
+      // Check that the logs contain all necessary fields
+      if (bridge.blockNumber && bridge.data) {
+        const hexData = bridge.data.startsWith('0x')
+          ? bridge.data.slice(2)
+          : bridge.data
+        // Check that the bridge is from the provided address (first block of data is sender)
+        if (
+          `0x${hexData.slice(24, 64)}`.toLowerCase() === address.toLowerCase()
+        ) {
+          const block = await getBlock(networkId, BigInt(bridge.blockNumber))
+          const blockTimestampDate = new Date(Number(block.timestamp) * 1000)
+          bridges.push({
+            amount: fromHex(`0x${hexData.slice(128, 192)}`, 'bigint'), // Amount is 3rd block of 32 bytes
+            tokenAddress: `0x${hexData.slice(88, 128)}`, // Token address is 2nd block of 32 bytes, skip first 12 to get 20 byte address
+            timestamp: blockTimestampDate,
+          })
+        }
+      } else {
+        console.log(
+          `Rhino bridge transaction missing required field, blockNumber: ${bridge.blockNumber}, data: ${bridge.data}`,
+        )
+      }
+    }
+  })
+  return bridges
+}
+
+export async function getTotalRevenueUsdFromBridges({
+  userBridges,
+  networkId,
+  startTimestamp,
+  endTimestampExclusive,
+}: {
+  userBridges: BridgeTransaction[]
+  networkId: NetworkId
+  startTimestamp: Date
+  endTimestampExclusive: Date
+}): Promise<number> {
+  if (userBridges.length === 0) {
+    return 0
+  }
+
+  let totalUsdContribution = 0
+
+  // For each bridge compute the USD contribution and add to the total
+  for (const bridge of userBridges) {
+    // Get the token decimals
+    const isNative = bridge.tokenAddress === zeroAddress
```

Comment:
if you have a link to some code in rhino that enforces this, it'd be helpful to add that as a comment :) i've usually see dapps use the 0xeeeeeee... address for native tokens, and use the zero address for minting and burning so i just found this convention to be a bit odd